Assets/Scripts/NetworkClient.cs

```diff
@@ -273,4 +312,61 @@ public bool IsConnected()
     {
         return mainWebSocket != null && mainWebSocket.State == WebSocketState.Open;
     }
+
+    private Dictionary<string, string> GetURLQueryParameters()
+    {
+        var output  = new Dictionary<string, string>();
+
+        string url = Application.absoluteURL;
+        if (url?.Length > 0)
+        {
+            var splitUrl = url.Split('?');
+            if (splitUrl.Length == 2)
+            {
+                var queryString = splitUrl[1];
+                var paramsCollection = HttpUtility.ParseQueryString(queryString);
+
+                foreach (var key in paramsCollection.AllKeys)
+                {
+                    output[key] = paramsCollection[key];
+                }
+            }
+        }
+
+        return output;
+    }
+
+    private void GetConnectionParameters(Dictionary<string, string> queryParams,                                   
```

Comment:
Nit on naming: looking at your callsite earlier, you are passing in connection params, not getting them. I'd call this `GetServerHostnameAndPort`.